ACHNBrowserUI/ACHNBrowserUI/views/turnips/charts/TurnipsChartView.swift

```diff
@@ -9,40 +9,48 @@
 import SwiftUI
 import Backend
 
+
 struct TurnipsChartView: View {
+    private struct ChartHeightPreferenceKey: PreferenceKey {
+        static var defaultValue: CGFloat?
+        static func reduce(value: inout CGFloat?, nextValue: () -> CGFloat?) {
+            if let newValue = nextValue() { value = newValue }
+        }
+    }
+    
     typealias PredictionCurve = TurnipsChart.PredictionCurve
     static let verticalLinesCount: CGFloat = 9
     var predictions: TurnipPredictions
     @Binding var animateCurves: Bool
     @Environment(\.presentationMode) var presentation
-    @State private var curvesFrame: CGRect = .zero
-    @State private var chartFrame: CGRect = .zero
-
+    
+    @State private var chartFrame: CGRect?
+    @State private var chartHeight: CGFloat?
+    
     var body: some View {
         VStack {
             TurnipsChartTopLegendView()
             HStack(alignment: .top) {
                 TurnipsChartVerticalLegend(predictions: predictions)
-                    .frame(width: 30, height: curvesFrame.height)
-                    .alignmentGuide(VerticalAlignment.top, computeValue: legendVerticalAlignment)
+                    .frame(width: 30, height: chartHeight)
+                    .padding(.top)
                 ScrollView(.horizontal, showsIndicators: false) {
                     chart.frame(width: 600, height: 500)
                 }
             }
         }
+        .onHeightPreferenceChange(ChartHeightPreferenceKey.self, storeValueIn: $chartHeight)
     }
-
+    
```

Comment:
Could you remove extra blank spaces? You could configure your Xcode to avoid that
![image](https://user-images.githubusercontent.com/2297556/81602232-1dbcb180-93c4-11ea-974d-42c089bc175e.png)

Reply:
Xcode doesn't seem to see a change here I got nothing to commit 😧